src/query.ts

```diff
@@ -204,231 +243,174 @@ export class Query {
-   *
-   * @param  {String}   field   The field to order by
-   *
-   * @param  {String}   order   Order literal to order by
-   *
-   */
-  public addSorting(field: string, order = Sort.ASC): void {
-    // This is so Prettier won't change it to const automagically
-    let sortParams: RequestSortOptions = [];
-    sortParams = this.params.sort || [];
-    this.removeSorting(field);
-    const obj: GenericObject = {};
-    obj[field] = order;
-    (sortParams as Array<RequestSortOptions>).push(obj);
-
-    this.params.sort = sortParams;
-  }
-
-  /**
-   * Sets all the sorting options at once
-   *
-   * @param  {Array}  sortings    The sortings in the order wanted
-   *
-   */
-  public setSorting(sortings: Array<RequestSortOptions>): void {
-    this.params.sort = sortings as RequestSortOptions;
+  public addSort(fieldOrList: InputSortValue[]): Query;
+  public addSort(fieldOrList: string, sortType?: string): Query;
+  public addSort(fieldOrList: string | InputSortValue[], sortType: string = OrderType.ASC): Query {
+    if (typeof fieldOrList === 'string') {
+      this._sort.set(fieldOrList, this._validateOrderType(sortType));
+    } else {
+      for (const value of fieldOrList) {
+        if (typeof value === 'string') {
+          this._sort.set(value, OrderType.ASC);
+        } else {
+          const field = Object.keys(value)[0];
+          const sortType: string = Object.values(value)[0] as string;
+          this._sort.set(field, this._validateOrderType(sortType));
+        }
+      }
+    }
+    return this;
   }
 
   public hasSorting(field: string): boolean {
-    if (!this.params.sort) {
-      return false;
-    } else {
-      return this._findSortField(field, this.params.sort as Array<RequestSortOptions>) !== -1;
-    }
+    return this._sort.has(field);
   }
 
   /**
    * Sets the page value of the request, useful for pagination
    *
-   * @param  {Number}     page              The page we want to set
+   * @param  page - The page we want to set
+   *
+   * @returns Query
    *
    */
-  public page(page?: number): void {
-    if (page) {
-      this.setParameter('page', page);
+  public page(page?: number): Query {
+    if (typeof page === 'number' || typeof page === 'undefined') {
+      this._page = page;
     } else {
-      delete this.params.page;
+      throw new QueryValueError('Value error: Page value must be a number');
     }
+    return this;
   }
 
   /**
    * Advances the current page to the next one
    *
    */
   public nextPage(): void {
-    this.setParameter('page', (this.params.page || 1) + 1);
+    this._page = (this._page || 1) + 1;
   }
 
   /**
    * Sets the Results Per Page (rpp) parameter.
    *
-   * @param  {Number}   rpp   The results per page to set
+   * @param  rpp - The results per page to set
    *
+   * @returns Query
    */
-  public resultsPerPage(rpp?: number): void {
-    if (rpp) {
-      this.setParameter('rpp', rpp);
+  public resultsPerPage(rpp?: number): Query {
```

Comment:
Rename to `rpp`. This is another special parameter. There's a forced max value of `100`, so the number shouldn't be greater than that. And we have a default value of… ¿`20`?

I think the default should be configurable via a static property, something like `Query.defaultRpp = 20`, being that the default.

So, you would do something like with `page` here:

```typescript
const value: number = parseInt(rpp || Query.defaultRpp);
if (isNaN(value)) {
  throw new QueryValueError();
} else {
  this._rpp = Math.min(value, MAX_RPP); // not configurable
}
```

Reply:
I can't do that because rpp must be a string in parseInt

Reply:
In JS you can pass anything, I don't know if TS will complain. For instance:

```js
parseInt({a: 23}, 10); // NaN
```